tests/functional/test_data_classes.py

```diff
@@ -126,6 +126,155 @@ class DataClassSample(DictWrapper):
     assert event_source.items() == data.items()
 
 
+def test_dict_wrapper_str_no_property():
+    class DataClassSample(DictWrapper):
+        attribute = None
+
+        def function(self):
+            pass
+
+    event_source = DataClassSample({})
+    assert event_source._properties() == ["raw_event"]
+    assert event_source._str_helper() == {"raw_event": "[SENSITIVE]"}
+    assert event_source.__str__() == "{'raw_event': '[SENSITIVE]'}"
```

Comment:
I'm not sure if we need to test this after all what we expect as a result is that the string has the `sensitive fields` marked as `[SENSITIVE]` and the others with the plain value of the event. 
The expected behavior is tested in the line below.

I would like to hear your opinion here too @heitorlessa.

Reply:
I agree that it is a bit overkill in some aspects, as I'm testing Python standard library works:
- `str(dict({})) == "{}"`
- `str(object)` actually calls `object.__str__(self)`

I might still want to ensure that `object.__str__()` is calling `object._str_helper()` - happy to discuss that further.

Happy to remove those checks if they're considered useless.

Reply:
The intent is fine but we don't need to be overzealous because we should be testing what the end customer will be using - `str(event_source`), of which you do perfectly ;)

Reply:
Thank you for the feedback, @heitorlessa! 
Please remove those checks @neilramsay.

Reply:
I've removed the `assert event_source.__str__() == "{'raw_event': '[SENSITIVE]'}"` check across all the new tests.
I've kept the `._properties()` and `._str_helper()` checks in as these ensure sensitive properties are being respected, and recursion works.

Reply:
I made some modifications to the tests. Testing private scope variables might be confusing here. 
As @heitorlessa mentioned, if the result of asserting `str(event_source) == event_str` is what the test expect, then I think everything worked as expected.